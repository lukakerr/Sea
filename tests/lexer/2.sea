;; Factorial function
fn factorial {num n} -> num (
  if {n < 0} (
    ret -1
  ) else (
    num r = 1

    while {n > 0} (
      r *= n
      n -= 1
    )

    ret r
  )
)