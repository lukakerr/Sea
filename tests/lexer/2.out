[Kwd Fn,Identifier "factorial",LBrace,DataType N,Identifier "n",RBrace,Arrow,DataType N,LParen,Kwd If,LBrace,Identifier "n",Operator Lt,Num 0,RBrace,LParen,Kwd Ret,Operator Minus,Num 1,RParen,Kwd Else,LParen,DataType N,Identifier "r",Operator Eq,Num 1,Kwd While,LBrace,Identifier "n",Operator Gt,Num 0,RBrace,LParen,Identifier "r",Operator TimesEq,Identifier "n",Identifier "n",Operator MinusEq,Num 1,RParen,Kwd Ret,Identifier "r",RParen,RParen]
